packages/workflow/src/components/workflow-menu.js

```diff
@@ -0,0 +1,302 @@
+import { Icon, search as inputIcon } from '@wordpress/icons';
+import { executeAbility, store as abilitiesStore } from '@wordpress/abilities';
+
+const inputLabel = __( 'Run abilities and workflows' );
+
+function WorkflowInput( { isOpen, search, setSearch, abilities } ) {
+	const workflowMenuInput = useRef();
+	const _value = useCommandState( ( state ) => state.value );
+	const selectedItemId = useMemo( () => {
+		// Find the ability whose label matches the selected value
+		const ability = abilities.find( ( a ) => a.label === _value );
+		return ability?.name;
+	}, [ _value, abilities ] );
+	useEffect( () => {
+		// Focus the workflow palette input when mounting the modal.
+		if ( isOpen ) {
+			workflowMenuInput.current.focus();
+		}
+	}, [ isOpen ] );
+	return (
+		<Command.Input
+			ref={ workflowMenuInput }
+			value={ search }
+			onValueChange={ setSearch }
+			placeholder={ inputLabel }
+			aria-activedescendant={ selectedItemId }
+		/>
+	);
+}
+
+/**
+ * @ignore
+ */
+export function WorkflowMenu() {
+	const { registerShortcut } = useDispatch( keyboardShortcutsStore );
+	const [ search, setSearch ] = useState( '' );
+	const [ isOpen, setIsOpen ] = useState( false );
+	const [ abilityOutput, setAbilityOutput ] = useState( null );
+	const [ isExecuting, setIsExecuting ] = useState( false );
+	const containerRef = useRef();
+
+	const abilities = useSelect( ( select ) => {
+		const allAbilities = select( abilitiesStore ).getAbilities();
+		return allAbilities || [];
+	}, [] );
+
+	const filteredAbilities = useMemo( () => {
+		if ( ! search ) {
+			return abilities;
+		}
+		const searchLower = search.toLowerCase();
+		return abilities.filter(
+			( ability ) =>
+				ability.label?.toLowerCase().includes( searchLower ) ||
+				ability.name?.toLowerCase().includes( searchLower )
+		);
+	}, [ abilities, search ] );
+
+	// Focus container when output is shown so it can receive keyboard events
+	useEffect( () => {
+		if ( abilityOutput && containerRef.current ) {
+			containerRef.current.focus();
+		}
+	}, [ abilityOutput ] );
+
+	useEffect( () => {
+		registerShortcut( {
+			name: 'core/workflows',
+			category: 'global',
+			description: __( 'Open the workflow palette.' ),
+			keyCombination: {
+				modifier: 'primary',
+				character: 'j',
+			},
+		} );
+	}, [ registerShortcut ] );
+
+	useShortcut(
+		'core/workflows',
+		/** @type {import('react').KeyboardEventHandler} */
+		( event ) => {
+			// Bails to avoid obscuring the effect of the preceding handler(s).
+			if ( event.defaultPrevented ) {
+				return;
+			}
+
+			event.preventDefault();
+			setIsOpen( ! isOpen );
+		},
+		{
+			bindGlobal: true,
+		}
+	);
+
+	const closeAndReset = () => {
+		setSearch( '' );
+		setIsOpen( false );
+		setAbilityOutput( null );
+		setIsExecuting( false );
+	};
+
+	const goBack = () => {
+		setAbilityOutput( null );
+		setIsExecuting( false );
+		setSearch( '' );
+	};
+
+	const handleExecuteAbility = async ( abilityName ) => {
```

Comment:
On line 264 we are passing handleExecuteAbility( ability.name ), why not pass the full ability and avoid the find here?